bindings/ios/MEGABackgroundMediaUpload.h

```diff
@@ -0,0 +1,70 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ *
+ * @copyright Simplified (2-clause) BSD License.
+ *
+ * You should have received a copy of the license along with this
+ * program.
+ */
+
+#import <Foundation/Foundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface MEGABackgroundMediaUpload : NSObject
+
+/**
+ * @brief Encrypt the file or a portion of it
+ *
+ * Call this function once with the file to be uploaded.  It uses mediainfo to extract information that will
+ * help the webclient show or play the file in various browsers.  The information is stored in this object
+ * until the whole operation completes.   The encrypted data is stored in a new file.
+ *
+ * In order to save space on mobile devices, this function can be called in such a way that the last portion
+ * of the file is encrypted (to a new file), and then that last portion of the file is removed by file truncation.
+ * That operation can be repeated until the file is completely encrypted, and only the encrypted version remains,
+ * and takes up the same amount of space on the device.   The size of the portions must first be calculated by using
+ * the 'adjustsizeonly' parameter, and iterating from the start of the file, specifying the approximate sizes of the portions.
+ *
+ * Encryption is done by reading small pieces of the file, encrypting them, and outputting to the new file,
+ * so that RAM usage is not excessive.
+ *
+ * @param inputFilePath The file to encrypt a portion of (and the one that is ultimately being uploaded).
+ * @param start The index of the first byte of the file to encrypt
+ * @param length The number of bytes of the file to encrypt.  The function will round this value up by up to 1MB to fit the
+ *        MEGA internal chunking algorithm.  The number of bytes acutally encrypted and stored in the new file is the updated number.
+ * @param outputFilePath The name of the new file to create, and store the encrypted data in.
+ * @param urlSuffix The function will update the string passed in.  The content of the string must be appended to the URL
+ *        when this portion is uploaded.
+ * @param adjustsSizeOnly If this is set true, then encryption is not performed, and only the length parameter is adjusted.
+ *        This feature is to enable precalculating the exact sizes of the file portions for upload.
+ */
+- (BOOL)encryptFileAtPath:(NSString *)inputFilePath startPosition:(int64_t)start length:(unsigned *)length outputFilePath:(NSString * _Nullable)outputFilePath urlSuffix:(NSString * _Nullable * _Nullable)urlSuffix adjustsSizeOnly:(BOOL)adjustsSizeOnly;
+
+/**
+ * @brief Retrieves the value of the uploadURL once it has been successfully requested via requestBackgroundUploadURLWithFileSize
+ */
+- (NSString *)uploadURLString;
+
+/**
+ * @brief Turns the data stored in this object into a binary data.
+ *
+ * The object can then be recreated via MegaApi::backgroundMediaUploadResume and supplying the same binary.
```

Comment:
Maybe with the [MEGASdk resumeBackgroundMediaUploadBySerializedData:] ?

Reply:
Good catch. I have changed it to 

> resumeBackgroundMediaUploadBySerializedData in MEGASdk

MEGASdk is a class, here I think it is more appropriate to introduce the method rather than giving a real coding syntax.

Reply:
It is ok, but the Objective-C method instead the c++ method